src/clj/puppetlabs/services/jruby/jruby_puppet_core.clj

```diff
@@ -105,6 +108,25 @@
     (update-in [:max-active-instances] #(or % (default-pool-size (ks/num-cpus))))
     (update-in [:max-requests-per-instance] #(or % 0))))
 
+(def facter-jar
+  "Well-known name of the facter jar file"
+  "facter.jar")
+
+(schema/defn ^:always-validate
+  add-facter-jar-to-system-classloader
+  "Searches the ruby load path for a file whose name matches that of the
+  facter jar file.  The first one found is added to the system classloader's
+  classpath.  If no match is found, an info message is written to the log
+  but no failure is returned"
+  [ruby-load-path :- (schema/both (schema/pred vector?) [schema/Str]) ]
+  (if-let [facter-jar (first
+                        (filter fs/exists?
+                          (map #(fs/file % facter-jar) ruby-load-path)))]
+    (do
+      (log/debugf "Adding facter jar to classpath from: %s" facter-jar)
+      (ks-classpath/add-classpath facter-jar))
```

Comment:
One thing which bothers me about this is that reflection has to be used under the hood.  This is needed because `add-classpath` calls through to `URLClassLoader.addURL`, which is a protected method.  See https://github.com/tobias/dynapath/blob/v0.2.3/src/dynapath/defaults.clj#L17-L21.  Not sure there's any way to avoid this, though, without having the path to `facter.jar` be appended to the `java -cp` argument on the command line, which we specifically did not want to do for this work.